magefile.go

```diff
@@ -933,23 +930,29 @@ func packageAgent(platforms []string, packagingFn func()) {
 				"pf-host-agent",
 			}
 
-			ctx := context.Background()
+			packageRequests := []tools.PackageRequest{}
+			downloadVersion := packageVersion
+			if _, ok := os.LookupEnv(snapshotEnv); ok {
+				downloadVersion += "-SNAPSHOT"
+			}
 			for _, binary := range externalBinaries {
 				for _, platform := range platforms {
 					reqPackage := platformPackages[platform]
 					targetPath := filepath.Join(archivePath, reqPackage)
 					os.MkdirAll(targetPath, 0755)
-					newVersion, packageName := getPackageName(binary, packageVersion, reqPackage)
-					err := fetchBinaryFromArtifactsApi(ctx, packageName, binary, newVersion, targetPath)
-					if err != nil {
-						if strings.Contains(err.Error(), "object not found") {
-							fmt.Printf("Downloading %s: unsupported on %s, skipping\n", binary, platform)
-						} else {
-							panic(fmt.Sprintf("fetchBinaryFromArtifactsApi failed for %s on %s: %v", binary, platform, err))
-						}
-					}
+					_, packageName := getPackageName(binary, downloadVersion, reqPackage)
+					packageRequests = append(packageRequests, tools.PackageRequest{Name: packageName, TargetPath: targetPath})
 				}
 			}
+
+			ctx := context.Background()
+
+			artifactSnapshotClient := tools.NewArtifactSnapshotClient()
+			err = artifactSnapshotClient.DownloadPackages(ctx, packageRequests, downloadVersion)
```

Comment:
[NIT]
as the context is being created just to satisfy `artifactSnapshotClient.DownloadPackages` it seems to be a good place to use `context.TODO()`
```suggestion
			artifactSnapshotClient := tools.NewArtifactSnapshotClient()
			err = artifactSnapshotClient.DownloadPackages(context.TODO(), packageRequests, downloadVersion)
```